libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -212,29 +207,40 @@ proc bootstrap*(
     try:
       await kad.switch.connect(b.peerId, b.addrs)
       debug "connected to bootstrap peer", peerId = b.peerId
-    except CatchableError as e:
-      error "failed to connect to bootstrap peer", peerId = b.peerId, error = e.msg
-
-    try:
-      let msg = await kad.sendFindNode(b.peerId, b.addrs, kad.rtable.selfId).wait(
-        chronos.seconds(5)
-      )
-      for peer in msg.closerPeers:
-        let p = PeerId.init(peer.id).tryGet()
-        discard kad.rtable.insert(p)
+    except DialFailedError as e:
+      # at some point will want to bubble up a Result[void, SomeErrorEnum]
+      error "failed to dial to bootstrap peer", peerId = b.peerId, error = e.msg
+      return
```

Comment:
if a bootstrap node is unreachable, the client should skip it and try the next one. It should be expected to include multiple bootstrap addresses so the failure of any single one doesn’t block the bootstrapping process

```suggestion
      continue
```